chpl/chpl-service/src/main/java/gov/healthit/chpl/svap/manager/SvapManager.java

```diff
@@ -1,24 +1,207 @@
+        return svapDao.create(svap);
+    }
+
+    private void deleteSvap(Svap svap) throws EntityRetrievalException {
+        svapDao.remove(svap);
+    }
+
+    private void deleteCriteriaRemovedFromSvap(Svap updatedSvap, Svap originalSvap) {
+        getCriteriaRemovedFromSvap(updatedSvap, originalSvap).stream()
+                .forEach(crit -> svapDao.removeSvapCriteriaMap(updatedSvap, crit));
+    }
+
+    private void deleteAllCriteriaFromSvap(Svap svap) {
+        svap.getCriteria().stream()
+                .forEach(crit -> svapDao.removeSvapCriteriaMap(svap, crit));
+    }
+
+    private void addNewCriteriaForExistingSvap(Svap updatedSvap, Svap originalSvap) {
+        getCriteriaAddedToSvap(updatedSvap, originalSvap).stream()
+                .forEach(crit -> svapDao.addSvapCriteriMap(updatedSvap, crit));
+    }
+
+    private void addNewCriteriaForNewSvap(Svap svap, List<CertificationCriterion> criteria) {
+        criteria.stream()
+                .forEach(crit -> svapDao.addSvapCriteriMap(svap, crit));
+    }
+
+    private void validateForDelete(Svap svap) throws ValidationException {
+        List<CertifiedProductDetailsDTO> listings = svapDao.getCertifiedProductsBySvap(svap);
+        if (listings.size() > 0) {
+            String message = errorMessageUtil.getMessage("svap.delete.listingsExist",
+                    listings.size(),
+                    listings.stream()
+                            .map(listing -> listing.getChplProductNumber())
+                            .collect(Collectors.joining(", ")));
+            ValidationException e = new ValidationException(message);
+            throw e;
+        }
+    }
+
+    private void validateForEdit(Svap updatedSvap, Svap originalSvap) throws ValidationException {
+        Set<String> messages = new HashSet<String>();
+
+        if (StringUtils.isEmpty(updatedSvap.getApprovedStandardVersion())) {
+            messages.add(errorMessageUtil.getMessage("svap.edit.emptyApprovedStandardVersion"));
+        }
+
+        if (StringUtils.isEmpty(updatedSvap.getRegulatoryTextCitation())) {
+            messages.add(errorMessageUtil.getMessage("svap.edit.emtptyRegulatoryTextCitation"));
+        }
+
+        if (updatedSvap.getCriteria().size() == 0) {
+            messages.add(errorMessageUtil.getMessage("svap.edit.noCriteria"));
+        }
+
+        //If there are removed criteria, make sure there are no listings attesting to SVAP/criteria
+        getCriteriaRemovedFromSvap(updatedSvap, originalSvap).stream()
+                .forEach(crit -> {
+                    List<CertifiedProductDetailsDTO> listings = svapDao.getCertifiedProductsBySvapAndCriteria(originalSvap, crit);
+                    if (listings.size() > 0) {
+                        messages.add(errorMessageUtil.getMessage("svap.edit.deletedCriteria.listingsExist",
+                                crit.getNumber() + (crit.getTitle().indexOf("Cures Update") == -1 ? "" : " (Cures Update)"),
```

Comment:
I think there is a criteria number formatter somewhere that does this sort of thing and we use in lots of the validators where we are printing out criterion numbers. Can you dig that up and use it here?